packages/grafana-ui/src/components/AutoSaveField/AutoSaveField.story.tsx

```diff
@@ -0,0 +1,208 @@
+    reject();
+  });
+};
+const themeOptions = [
+  { value: '', label: 'Default' },
+  { value: 'dark', label: 'Dark' },
+  { value: 'light', label: 'Light' },
+  { value: 'system', label: 'System' },
+];
+enum ThemeSwitchOption {
+  Light = 'light',
+  Dark = 'dark',
+}
+
+export const Basic: Story = (args) => {
+  const [inputValue, setInputValue] = useState('');
+  return (
+    <AutoSaveField onFinishChange={args.inputSuccessful ? getSuccess : getError} {...args}>
+      {(onChange) => (
+        <Input
+          value={inputValue}
+          onChange={(e) => {
+            const value = e.currentTarget.value;
+            onChange(value);
+            setInputValue(e.currentTarget.value);
+          }}
+        />
+      )}
+    </AutoSaveField>
+  );
+};
+
+Basic.args = {
+  required: false,
+  label: 'Input saving value automatically',
+  inputSuccessful: false,
+};
+
+export const AllComponents: Story = (args) => {
+  const [selected, setSelected] = useState('');
+  const [checkBoxTest, setCheckBoxTest] = useState(false);
+  const [textAreaValue, setTextAreaValue] = useState('');
+  const [inputTextValue, setInputTextValue] = useState('');
+  const [switchTest, setSwitchTest] = useState(false);
+
+  return (
+    <div>
+      <AutoSaveField onFinishChange={args.inputSuccessful ? getSuccess : getError} label="Text as a child" {...args}>
+        {(onChange) => (
+          <Input
+            value={inputTextValue}
+            onChange={(e) => {
+              const value = e.currentTarget.value;
+              onChange(value);
+              setInputTextValue(e.currentTarget.value);
+            }}
+          />
+        )}
+      </AutoSaveField>
+      <AutoSaveField onFinishChange={args.inputSuccessful ? getSuccess : getError} label="Select as child" {...args}>
+        {(onChange) => (
+          <Select
+            options={themeOptions}
+            value={args.weekPickerValue}
+            onChange={(v) => {
+              onChange(v.value);
+            }}
+          />
+        )}
+      </AutoSaveField>
+      <AutoSaveField
+        onFinishChange={args.inputSuccessful ? getSuccess : getError}
+        label="RadioButtonGroup as a child"
+        {...args}
+      >
+        {(onChange) => (
+          <RadioButtonGroup
+            options={themeOptions}
+            value={selected}
+            onChange={(themeOption) => {
+              setSelected(themeOption);
+              onChange(themeOption);
+            }}
+          />
+        )}
+      </AutoSaveField>
+      <AutoSaveField
+        onFinishChange={args.inputSuccessful ? getSuccess : getError}
+        label="Checkbox as a child"
+        {...args}
+      >
+        {(onChange) => (
+          <Checkbox
+            label="Checkbox test"
+            description="This is a checkbox input"
+            name="checkbox-test"
+            value={checkBoxTest}
+            onChange={(e) => {
+              const value = e.currentTarget.checked.toString();
```

Comment:
you don't pass a generic type to the `AutoSaveField` component which is why you have to stringify here...

if you pass `Boolean` as the generic type, you can simply call `onChange` with `e.currentTarget.checked`